ADR: Home Assistant Supervised as supported installation method (#394)
home-assistant/architecture: adr/0014-home-assistant-supervised.md

# 0012. Installation method: Home Assistant Container

Date: 2020-06-03

## Status

Accepted

## Context

Define a supported installation method as per [ADR-0012](https://github.com/home-assistant/architecture/blob/master/adr/0012-define-supported-installation-method.md).

## Decision

> This installation method provides the full Home Assistant experience on a regular operating system. This means, all components from the Home Assistant method are used, except for the Home Assistant Operating System. We don’t support using the host system for other applications than the Supervisor. You need to understand that the Supervisor is not just an application, it is a full appliance.

Author: I've reworded it to make it more part of the definition.

> By not using the Home Assistant Operating System, the user is responsible for making sure that all required components are installed and maintained. Required components and their versions will change over time. We are not responsible for issues that come up with your host or wrong configuration.

Reviewer: > or wrong configuration
I would suggest
> or corroecting a wrong configuration

> This method is considered advanced and should only be used if one is an expert in managing a Linux operating system, Docker and networking.

Reviewer 2: Don't be too specific here. The details are out-lined after.

```suggestion
This method is considered advanced and should only be used if one is an expert in managing an operating system, containerization solution and networking.
```

Reviewer 3: We can generalize this that way for a supervised installation, as it can't be run on another operating system or containerization system.

### Supported Operating System, System dependencies and versions

> Docker CE (Community Edition) is the only supported containerization method for Home Assistant Supervised. We only support FSH 3.0 on the host file system.

Reviewer 4: Is this meant to be [FHS](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/index.html)?

Reviewer 3: Yes, that is a typo.

- Docker CE >= 19.03
- Systemd >= 239
- NetworkManager >= 1.18.0
- Avahi >= 0.7
- AppArmor == 2.13.x
- Debian Linux Debian 10 aka Buster

> Only the above-listed version of Debian Linux is supported for running this installation method. When a new major version of Debian is released, the previous major version is dropped, with a deprecation time of 4 months. An exception to this rule occurs if the new version does not solve the requirements of the Supervisor.

Reviewer: > does not solve the requirements of 
I'd suggest something more like:
> does not meet the requirements of
or
> does not provide the requirements for

### Additional supported conditions

This installation method can easily be broken if one manages the operating system incorrectly. Therefore, the following additional conditions apply:

- The operating system is dedicated to running Home Assistant Supervised.
- All system dependencies are installed according to the manual.

> - No additional software, outside of the Home Assistant ecosystem, is installed.

Author: This one will need some extra explaining. It will be along the lines: 

The reason for this requirement is that the Supervisor turns your system into an appliance: a system with the specific task of running Home Assistant. This is not an "app" that you install. Your system becomes Home Assistant Supervised.

This means that the supervisor will expect to be able to clean and correct the system to ensure smooth operations. Examples:

 - If a config can't be parsed, we should be able to reset it to defaults.
 - cleanup tasks will remove all Docker containers that are not running

- Docker needs to be correctly configured to use overlayfs2 storage and journald as the logging driver.
- NetworkManager is installed but not actively used.

In case any abnormality is detected that prevents Home Assistant from functioning, the Home Assistant Supervisor will report this to the user and block updates to prevent installations from breaking.

### Required Expertise

- **Installation**
  The user first needs to install a Linux-based operating system (i.e. Debian). They then need to run the installer script.

* **Start when the system is started:** done by the installer
* **Run with full network access:** done by the installer
* **Access USB devices:** done by the installer

* **Maintenance Home Assistant installation**
  Home Assistant can be maintained from the supervisor. Includes rollback when the update fails.

  - **Python upgrades:** Included in Home Assistant updates
  - **Installing Python dependencies:** Included in Home Assistant updates
  - **Updating Home Assistant:** Via the UI

- **Maintenance Operating System**

>   The user is responsible for maintaining the operating system. Since it is a supervised installation, the user is also responsible for updating the components that are required by the supervisor. The user is also responsible for installing or changing anything on their system that will interfere with the Supervised installation. Examples are software that will update Docker containers managed by the Supervisor.

Reviewer: > The user is also responsible for installing or changing anything on their system that will interfere 
Makes it sound like they should do that ;)
> The user is also responsible for not installing or changing anything on their system that will interfere

- AppArmor == 2.13.x
- Debian Linux Debian 10 aka Buster

Only the above-listed version of Debian Linux is supported for running this installation method. When a new major version of Debian is released, the previous major version is dropped, with a deprecation time of 4 months. An exception to this rule occurs if the new version does not meet the requirements of the Supervisor.

### Additional supported conditions

This installation method can easily be broken if one manages the operating system incorrectly. Therefore, the following additional conditions apply:

- The operating system is dedicated to running Home Assistant Supervised.
- All system dependencies are installed according to the manual.
- No additional software, outside of the Home Assistant ecosystem, is installed.
- Docker needs to be correctly configured to use overlayfs2 storage and journald as the logging driver.
- NetworkManager is installed but not actively used.

In case any abnormality is detected that prevents Home Assistant from functioning, the Home Assistant Supervisor will report this to the user and block updates to prevent installations from breaking.

### Required Expertise

- **Installation**

>   The user first needs to install a Linux-based operating system (i.e. Debian). They then need to run the installer script.

Reviewer 5: ```suggestion
  The user first needs to install a Linux-based operating system (i.e. Debian). The user then needs to run the installer script.
```

Reviewer 6: If the only supported version is Debian then shouldn't this read "The user first needs to install Debian 10"

* **Start when the system is started:** done by the installer
* **Run with full network access:** done by the installer
* **Access USB devices:** done by the installer

> * **Maintenance Home Assistant installation**

Reviewer 5: ```suggestion
* **Maintaining the Home Assistant installation**
```

>   Home Assistant can be maintained from the Supervisor. Includes rollback when the update fails.

Reviewer 5: ```suggestion
  Home Assistant can be maintained from the Supervisor. This includes a rollback when the update fails.
```

  - **Python upgrades:** Included in Home Assistant updates
  - **Installing Python dependencies:** Included in Home Assistant updates
  - **Updating Home Assistant:** Via the UI

> - **Maintenance Operating System**

Reviewer 5: ```suggestion
- **Maintaining the Operating System**
```
